packages/x-components/src/components/__tests__/base-extra-params.spec.ts

```diff
@@ -0,0 +1,154 @@
+import { mount } from '@vue/test-utils';
+import Vue, { VueConstructor } from 'vue';
+import Component from 'vue-class-component';
+import { ComponentOptions } from 'vue/types/umd';
+import { Dictionary } from '../../utils';
+import { ExtraParamXEvent } from '../../wiring';
+import { XProvide } from '../decorators/injection.decorators';
+import BaseExtraParams from '../base-extra-params.vue';
+
+describe('testing extra params component', () => {
+  const emitSpy = jest.fn();
```

Comment:
Using globals is unnecessary. It is better and easier to create a new jest function mock for each test, and return it as part of the test API. I would just name it `emit` in that case. Then you can avoid the `beforeEach` loop.